libs/shared/directives/src/validate-input.directive.ts

```diff
@@ -0,0 +1,65 @@
+import { Directive, ElementRef, inject, OnInit, OnDestroy, Renderer2 } from '@angular/core';
+import { AbstractControl, NgControl } from '@angular/forms';
+import { SubscriptionHandler } from '@keira/shared/utils';
+import { debounceTime, distinctUntilChanged } from 'rxjs';
+import { ValidationService } from '@keira/shared/common-services';
+
+@Directive({
+  selector: '[keiraInputValidation]',
+  standalone: true,
+})
+export class InputValidationDirective extends SubscriptionHandler implements OnInit, OnDestroy {
+  private readonly el: ElementRef = inject(ElementRef);
+  private readonly renderer: Renderer2 = inject(Renderer2);
+  private readonly ngControl: NgControl = inject(NgControl);
+  private readonly validationService = inject(ValidationService);
+
+  private errorDiv: HTMLElement | null = null;
+
+  ngOnInit(): void {
+    const control = this.ngControl.control;
+
+    if (!control) {
+      return;
+    }
+
+    this.validationService.setControlValidity(this, control.valid);
+
+    this.subscriptions.push(
+      control.statusChanges?.pipe(distinctUntilChanged(), debounceTime(500)).subscribe(() => {
+        this.updateErrorMessage(control);
+        this.validationService.setControlValidity(this, control.valid);
+      }),
+    );
+  }
+
+  override ngOnDestroy(): void {
+    this.validationService.removeControl(this);
+    super.ngOnDestroy();
+  }
+
+  private updateErrorMessage(control: AbstractControl | null): void {
+    if (this.errorDiv) {
+      this.renderer.removeChild(this.el.nativeElement.parentNode, this.errorDiv);
+      this.errorDiv = null;
+    }
+
+    if (control?.invalid) {
+      control.markAsTouched();
+    }
+
+    if (control?.touched && control?.invalid && control.errors) {
+      const parent = this.el.nativeElement.parentNode;
+      if (parent) {
+        this.errorDiv = this.renderer.createElement('div');
+        this.renderer.addClass(this.errorDiv, 'error-message');
```

Comment:
I think that I wrote this, but I am not sure if we have a class `error-message`, if not, we could remove this line of code

Reply:
We dont. But the tests use it to get the div in order to verify the correct message.

Reply:
I would use an id instead of a class for this